pkg/restore-util/split.go

```diff
@@ -46,43 +45,54 @@ func NewRegionSplitter(client Client) *RegionSplitter {
 // tableRules includes the prefix of a table, since some ranges may have a prefix with record sequence or index sequence.
 // note: all ranges and rewrite rules must have raw key.
 func (rs *RegionSplitter) Split(ctx context.Context, ranges []Range, rewriteRules *RewriteRules) error {
-	var wg sync.WaitGroup
 	rangeTree, ok := newRangeTreeWithRewrite(ranges, rewriteRules)
 	if !ok {
 		return errors.Errorf("ranges overlapped: %v", ranges)
 	}
-	err := rs.splitByRewriteRules(ctx, &wg, rewriteRules.Data)
+	scatterRegions, err := rs.splitByRewriteRules(ctx, rewriteRules.Data)
 	if err != nil {
 		return errors.Trace(err)
 	}
 	rangeTree.Ascend(func(rg *Range) bool {
 		if rg == nil {
 			return false
 		}
-		err = rs.maybeSplitRegion(ctx, rg, &wg)
+		var newRegion *RegionInfo
+		newRegion, err = rs.maybeSplitRegion(ctx, rg)
+		if err == nil {
+			if newRegion != nil {
+				scatterRegions = append(scatterRegions, newRegion)
+			}
+		}
 		return err == nil
```

Comment:
```go
if err != nil {
    return false
}
if newRegion != nil {
    scatterRegions = append(scatterRegions, newRegion)
}
return true
```

Reply:
oh, I forget to modify it after removing debug log.😂️